### PR TITLE
Fix node path

### PR DESCRIPTION
**Pull request by airportyh**
Why
===

Bug:
1. install node module in blank repl
2. node binary not found

Reason: I set the `PATH` env var in the module, overriding the `PATH` generated by lib.makeBinPath

This is not the first time I made this mistake, so to prevent future mistakes, I merged the PATHs automatically.

What changed
============

Added function to automatically merge the `PATH` variable if one is specified by the module.

Test plan
=========
```
nix build .#modules.'"nodejs-18"'
cat result | jq .env.PATH
```
See that the output has 2 entries under $REPL_HOME  and one that comes from /nix/store

Rollout
=======

_Describe any procedures or requirements needed to roll this out safely (or check the box below)_

- [ ] This is fully backward and forward compatible

### CODE REVIEW

File: pkgs/moduleit/module-definition.nix [hunk 1]
```diff
@@ -567,14 +567,18 @@ in
 
     replit.buildModule =
       let
-
+        envWithMergedPath = (env: path:
+          let packagesPath = lib.makeBinPath config.packages;
+          in
+          if builtins.hasAttr "PATH" env
+          then env // { PATH = "${env.PATH}\:${packagesPath}"; }
```
Comment (ryantm) [inline pkgs/moduleit/module-definition.nix:574]:
```suggestion
          then env // { PATH = "${env.PATH}:${packagesPath}"; }
```
Is there a reason to escape the colon? I tried it either way and it seems the same...

**Review by ryantm — APPROVED** [review]

File: pkgs/moduleit/module-definition.nix (hunk 1 above)
Comment (airportyh) [reply pkgs/moduleit/module-definition.nix:574]: Weird, it was not working for me with it. Let me try again.

Comment (ryantm) [reply pkgs/moduleit/module-definition.nix:574]: lol, fine to keep it then, it doesn't break anything.
